### ISSUE TITLE
Approval system fails to recognize users that have already been approved

### ISSUE THREAD

**Issue by jaredlyon** [KingWaffleshnoz/JareBot#27]
![image](https://user-images.githubusercontent.com/29807461/89143674-41dde980-d500-11ea-86ff-83311404d8ff.png)
![image](https://user-images.githubusercontent.com/29807461/89143695-51f5c900-d500-11ea-8c10-61398957d061.png)

Current code: 
```js
if (msg.channel.id == bot.config.welcomeChannel) {
            if (target != null) {
                if (!target.roles) {
                    var logEmbed = new Discord.MessageEmbed()
                        .setAuthor(msg.author.username, msg.author.avatarURL())
                        .addField('Member approved:', yup + ` **${target.username.toString()}#${target.discriminator} (${target.id}) was approved.**`)
                        .setFooter(bot.user.username, bot.user.avatarURL())
                        .setTimestamp()
                        .setColor(3447003);

                    msg.mentions.members.forEach(member => {
                        member.roles.add(msg.guild.roles.cache.find(role => role.name === "• Customers")).then(member => {
                            channel.send({
                                embed: logEmbed
                            })
                            log.send({
                                embed: logEmbed
                            })
                        });
                    })
                } else if (target.roles) {
                    msg.reply("user has already been approved!");
                } else {
                    msg.reply("something went wrong!");
                }
```

Executed properly, the targeted user should not trigger the primary if statement due to their roles.

**Comment by jaredlyon** [comment]
Attempts to remedy the issue using a `.cache.get` method to probe the target user for a specific role have been unsuccessful due to (presumably) an inability to prompt a cache refresh. Specific code examples with attached error logs may be attached to the thread at a later time.

**Comment by jaredlyon** [comment]
Discord.js v12 includes an update to presence and server member intents, so this issue may resolve itself on its own with minor tweaks. See issue #30 for more info.

**Comment by jaredlyon** [comment]
Updated the `if` statements used in commit ca9c6b2, but now we're getting this really weird kickback error that has no discernable cause:
<img width="928" alt="Screen Shot 2020-11-06 at 10 39 56 AM" src="https://user-images.githubusercontent.com/29807461/98386603-3faa9000-2005-11eb-8b5b-3206d2dde2ae.png">

**Comment by jaredlyon** [comment]
I ended up removing the role check entirely (1d87462); if duplicate logs become a problem then we'll revisit this issue in the future.

**Comment by jaredlyon** [comment]
It appears the same issue has also broken the `j!ban` command in moderation:
<img width="867" alt="Screen Shot 2020-11-09 at 8 17 24 AM" src="https://user-images.githubusercontent.com/29807461/98545889-01061700-224b-11eb-8437-97057e7f999c.png">

**Comment by jaredlyon** [comment]
Verified the above commit's (9c7e1ef) fix:
<img width="487" alt="Screen Shot 2020-11-09 at 10 22 58 AM" src="https://user-images.githubusercontent.com/29807461/98560271-70383700-225c-11eb-9fb9-b1f9f5cb16d6.png">

I'll try it for the original `j!approve` command next.

**Comment by jaredlyon** [comment]
Commit 35117e3 recreates the original undefined cache error throw:
<img width="867" alt="Screen Shot 2020-11-09 at 10 59 35 AM" src="https://user-images.githubusercontent.com/29807461/98564687-84326780-2261-11eb-9a0f-176fe1b70bb2.png">

**Comment by jaredlyon** [comment]
Fixed by 74d3189:
<img width="386" alt="Screen Shot 2020-11-09 at 12 22 07 PM" src="https://user-images.githubusercontent.com/29807461/98574775-328fda00-226d-11eb-82cf-87f921b09c8c.png">

The problem was that I had been using `.users` instead of `.members`...
